tests/unit_tests/sql/transpile_to_dialect_test.py

```diff
@@ -0,0 +1,271 @@
+
+
+def test_postgres_single_quote_escaping():
+    """PostgreSQL keeps SQL-92 standard: double single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "postgresql")
+    assert "O''Hara" in result
+    assert "name" in result
+
+
+def test_databricks_single_quote_escaping():
+    """Databricks uses backslash escaping for single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "databricks")
+    assert "name" in result
+    # Databricks uses backslash escaping
+    assert "O\\'Hara" in result or "O''Hara" in result
+
+
+def test_mysql_single_quote_escaping():
+    """MySQL uses backslash escaping for single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "mysql")
+    assert "name" in result
+    # MySQL may use backslash or double-quote escaping
+    assert (
+        "Connor" in result.replace("O''Hara", "Connor").replace("O\\'Hara", "Connor")
+        or "Hara" in result
+    )
+
+
+def test_sqlite_single_quote_escaping():
+    """SQLite uses SQL-92 standard: double single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "sqlite")
+    assert "O''Hara" in result
+    assert "name" in result
+
+
+def test_snowflake_single_quote_escaping():
+    """Snowflake handles single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "snowflake")
+    assert "name" in result
+    assert "Hara" in result
+
+
+def test_bigquery_single_quote_escaping():
+    """BigQuery handles single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "bigquery")
+    assert "name" in result
+    assert "Hara" in result
+
+
+def test_compound_filter_with_quotes():
+    """Test compound filters with quoted strings."""
+    sql = "(name = 'O''Hara' AND status = 'active')"
+    result = transpile_to_dialect(sql, "postgresql")
+    assert "Hara" in result
+    assert "active" in result
+    assert "AND" in result
+
+
+def test_like_with_special_chars():
+    """Test LIKE patterns with special characters."""
+    sql = "name LIKE '%O''Hara%'"
+    result = transpile_to_dialect(sql, "postgresql")
+    assert "Hara" in result
+    assert "LIKE" in result
+
+
+def test_ilike_case_insensitive():
+    """Test ILIKE for case-insensitive matching (PostgreSQL syntax)."""
+    sql = "name ILIKE '%O''Hara%'"
+
+    # PostgreSQL keeps ILIKE and '' escaping
+    pg_result = transpile_to_dialect(sql, "postgresql")
+    assert "ILIKE" in pg_result
+    assert "Hara" in pg_result
+
+    # MySQL converts ILIKE to LIKE (MySQL LIKE is case-insensitive by default)
+    mysql_result = transpile_to_dialect(sql, "mysql")
+    assert "LIKE" in mysql_result
+    assert "Hara" in mysql_result
+
+    # Databricks supports ILIKE
+    db_result = transpile_to_dialect(sql, "databricks")
+    assert "ILIKE" in db_result or "LIKE" in db_result
+    assert "Hara" in db_result
```

Comment:
Same thing here, let's use `@pytest.mark.parametrize` to break these down into individual tests.